src/sysusers/sysusers.c

```diff
@@ -1550,9 +1553,22 @@ static int parse_line(const char *fname, unsigned line, const char *buffer) {
 
                                 path_kill_slashes(i->uid_path);
                         } else {
+                                _cleanup_free_ char *uid = NULL, *gid = NULL;
+                                if (split_pair(resolved_id, ":", &uid, &gid) == 0) {
+                                        r = parse_gid(gid, &i->gid);
+                                        if (r < 0) {
+                                                log_error_errno(r, "Failed to parse GID: '%s': %m", id);
+                                                return -EBADMSG;
+                                        }
+                                        i->gid_set = true;
+                                        i->gid_existing_ok = true;
+                                        mfree(resolved_id);
+                                        resolved_id = uid;
+                                        uid = NULL;
```

Comment:
Please use `free_and_replace(resolved_id, uid);` instead of the above three lines.

Reply:
Thanks, fixed now. Nice helper, I was not aware of it.